### PR TITLE
Add an alternative "magic link sent" screen

### PR DESCRIPTION
**Pull request by 0nko**
This PR adds an alternative "magic link sent" screen, which will be used in WCAndroid A/B test for improving the login success rate.

image

### CODE REVIEW

File: WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginMagicLinkRequestFragment.java [hunk 1]
```diff
@@ -362,7 +362,7 @@ public void onAuthEmailSent(OnAuthEmailSent event) {
             if (mForceRequestAtStart) {
                 FragmentManager fragmentManager = getFragmentManager();
                 if (fragmentManager != null) {
-                    fragmentManager.popBackStackImmediate();
+                    fragmentManager.popBackStack();
```
Comment (0nko) [inline WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginMagicLinkRequestFragment.java:365]: This change fixes a race-condition, which would crash the app (`mLoginListener` is nulled) if there is no screen-transition animation between the "magic link request" screen and "magic link sent" screen.

**Comment by hichamboushaba** [comment]
@0nko just a suggestion, why not define the new fragment inside Woo directly? (I mean something similar to what we've done with other experiments, for example: https://github.com/woocommerce/woocommerce-android/pull/6891).
I think this would allow moving faster with iterations.

**Comment by nbradbury** [comment]
@0nko Looks good and I'll approve it, but I'll hold off merging until you have a chance to respond to @hichamboushaba's suggestion.

**Review by nbradbury — APPROVED** [review]
:shipit:

**Comment by annchichi** [comment]
Heya @0nko, thanks for this! 

I was looking for an illustration that fits the context here. I found the WooCommerce Admin has this letter illo, which may fit better, although the graph isn't that appealing. 😬 

CleanShot 2022-08-02 at 14 34 46@2x


No 1 Is it possible to change this letter illustration? 
No 2 Happy to reuse the copy. But I wonder if we can do a slight tweaking so we won't have one word left in the 2nd sentence. We can push "Check your Spam or Junk Mail folder" to the second line. Asset link [here](https://drive.google.com/drive/folders/1czL-BZsr5atndhM4D1l4KLzs8Ti9WalO?usp=sharing).

**Comment by 0nko** [comment]
Hey @annchichi, I've updated the UI:

image

**Comment by 0nko** [comment]
@hichamboushaba I suppose you're right, I automatically assumed it needed to be done here. But since it's already finished I'd say let's keep it and I'll do it in WCAndroid next time. 

cc: @nbradbury

**Pull request merged**